src/algorithms/calorimetry/CalorimeterHitDigi.cc

```diff
@@ -31,13 +29,8 @@ using namespace dd4hep;
 //   values here. This needs to be confirmed.
 
 
-
-//------------------------
-// AlgorithmInit
-//------------------------
-void CalorimeterHitDigi::AlgorithmInit(std::shared_ptr<spdlog::logger>& logger) {
-
-    // Assume all configuration parameter data members have been filled in already.
+void CalorimeterHitDigi::init(const dd4hep::Detector* detector, std::shared_ptr<spdlog::logger>& logger) {
+    m_log = logger;
```

Comment:
Best to assign detector pointer...
```suggestion
    m_detector = detector;
    m_log = logger;
```